scripts/question_answering/data_processing.py

```diff
@@ -0,0 +1,117 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=
+"""SQuAD data preprocessing."""
+__all__ = ['SQuADTransform', 'VocabProvider', 'preprocess_dataset']
+
+import re
+
+from mxnet import nd
+from mxnet.gluon.data import SimpleDataset
+
+from gluonnlp import Vocab, data
+from gluonnlp.data.batchify import Pad
+
+
+def preprocess_dataset(dataset, question_max_length, context_max_length):
```

Comment:
add one more preprocess method or class that can return the 
`record_id, original passage text, gold answer strings, token offsets into the original passage`
so that we can more easily evaluate the final net's output.

Reply:
Ok, will do.